src/app/header/header.component.ts

```diff
@@ -1,28 +1,38 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
 
 @Component({
     selector: 'app-header',
     templateUrl: './header.component.html',
     styleUrls: ['./header.component.scss']
 })
+
 export class HeaderComponent implements OnInit {
 
-    classSelector
-    constructor() { }
+    constructor(private router: Router) {}
 
     ngOnInit() {
-        // linkHighlight = function(activeLink) {
-        //     this.classSelector = document.querySelectorAll('.navigation');
-        //     this.classSelector.forEach(function(element) {
-        //         element.className = "hiddenClass";
-        //     });
-        //     this.classSelector[4].className = ".navigation end";
-        //
-        //     this.classSelector = document.querySelectorAll('.dropMenu');
-        //     this.classSelector.forEach(function(element) {
-        //         element.className = "hiddenClass";
-        //     });
-        // }
+        this.router.events.subscribe((event) => {
+            this.linkHighlight(this.router.url);
+        });
     }
 
+    linkHighlight = function(url) {
+        var links = document.querySelectorAll('.navigation');
```

Comment:
Do not user document.querySelector, it makes testing miserable.

http://angularjs.blogspot.hu/2016/04/5-rookie-mistakes-to-avoid-with-angular.html